docs/.linkcheckerrc

```diff
@@ -11,7 +11,7 @@ ignore=
   http.*amd.com/.*
   https.*dl.acm.org/doi/.*
   https.*tandfonline.com/.*
-ignorewarnings=http-robots-denied,https-certificate-error
+ignorewarnings=http-redirected,http-robots-denied,https-certificate-error
```

Comment:
It looks like as of the most recent release of `linkchecker` ([v10.3.0](https://github.com/linkchecker/linkchecker/releases/tag/v10.3.0)), HTTP redirects now throw a warning.

From this PR: https://github.com/linkchecker/linkchecker/pull/750

That's leading to this job failing even with 0 errors. ([example build](https://github.com/microsoft/LightGBM/actions/runs/6694632781/job/18188404568))

> That's it. 1087 links in 1115 URLs checked. 28 warnings found. 0 errors found.
> Error: Process completed with exit code 255.

In this PR, I'm proposing not considering the presence of such warnings to cause the job to fail. It'll take some time to go through the 28 warnings and fix them, and I'd prefer to get this job working again as soon as possible to catch truly broken links.

If reviewers agree, I'll put up a new issue documenting the desire to remove this filter and enforce no-redirects in this job again.

Reply:
I triggered `check-links` on this branch before adding this change to the config: https://github.com/microsoft/LightGBM/actions/runs/6766475624/job/18387635055

And despite there being 0 errors in the result ...

```text
That's it. 1088 links in 1116 URLs checked. 28 warnings found. 0 errors found.
Stopped checking at 2023-11-06 05:04:50+000 (6 minutes, 20 seconds)
```

... it's still failing

```text
Error: Process completed with exit code 255.
```

I suspect that maybe `linkchecker` returns a non-0 exit code if any warnings are found. The last successful run in this project was https://github.com/microsoft/LightGBM/actions/runs/6219774651/job/16878494852, and that ended with:

```text
That's it. 1089 links in 1125 URLs checked. 0 warnings found. 0 errors found.
```